File: frontend/src/types/TerminDTOFormWithFilmsDTOFormPlus.ts
import Termin from "./Termin.ts";
import FilmDTOFormPlus from "./FilmDTOFormPlus.ts";

// this one mirrors the Java class TerminDTOFormWithFilmsDTOFormPlus
// for entries on the ScreeningDetails component
export default interface TerminDTOFormWithFilmsDTOFormPlus {
    termin: Termin;
    mainfilms: FilmDTOFormPlus[];
    vorfilms: FilmDTOFormPlus[];
    terminGesamtlaufzeit: number
}

File: backend/src/main/java/org/pupille/backend/mysql/screening/TerminDTOFormWithFilmsDTOFormPlus.java
package org.pupille.backend.mysql.screening;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import org.pupille.backend.mysql.film.FilmDTOForm;
import org.pupille.backend.mysql.termin.TerminDTOForm;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
public class TerminDTOFormWithFilmsDTOFormPlus {
    private TerminDTOForm termin;
    private List<FilmDTOFormPlus> mainfilms; // vorfilm=false/null
    private List<FilmDTOFormPlus> vorfilms;   // vorfilm=true
}
